chainstate/src/detail/chainstateref.rs

```diff
@@ -639,19 +651,37 @@ impl<'a, S: BlockchainStorageWrite, O: OrphanBlocksMut> ChainstateRef<'a, S, O>
         spend_height: &BlockHeight,
         blockreward_maturity: &BlockDistance,
     ) -> Result<(), BlockError> {
-        let connected_txs =
-            self.make_cache_with_connected_transactions(block, spend_height, blockreward_maturity)?;
-        let consumed = connected_txs.consume()?;
+        let mut utxo_db = UtxosDB::new(&mut self.db_tx);
+        let mut block_undo: BlockUndo = Default::default();
```

Comment:
Why not make this a member of TransactionVerifier? Doesn't it get flushed with everything else on success?

Reply:
`transaction_verifier` doesn't work on the block level but with separate transactions. It lacks the context on how it is used - to connect or disconnect txs and whether it should set or del BlockUndo. So I decided to keep this logic here

Reply:
Still, when you're connecting transactions, you're doing them at the block level. Think and see if you can make this cleaner.